src/LaborMarket.sol

```diff
@@ -403,15 +505,17 @@ contract LaborMarket is LaborMarketInterface, NBadgeAuth {
         /// @dev Ensure the Request is no longer in the enforcement phase.
         require(
             block.timestamp >= request.enforcementExp,
-            'LaborMarket::claimRemainder: Enforcement deadline not passed'
+            "LaborMarket::claimRemainder: Enforcement deadline not passed"
         );
 
         /// @notice Get the signal state of the Request.
-        ServiceSignalState storage signalState = requestIdToSignalState[_requestId];
+        ServiceSignalState storage signalState = requestIdToSignalState[
+            _requestId
+        ];
 
         /// @notice Determine the amount of available Provider payments never redeemed.
         pTokenProviderSurplus =
-            (request.providerLimit - signalState.providersArrived) *
+            (request.providerLimit - requestIdToProviders[_requestId].length()) *
```

Comment:
requestIdToProviders has a provider removed from the set upon a successful claim:

```
            /// @notice Recover the address by truncating the Submission id.
            address provider = address(uint160(_submissionId));

            /// @notice Remove the Submission from the list of Submissions.
            /// @dev This is done before the transfer to prevent reentrancy attacks.
            bool removed = requestIdToProviders[_requestId].remove(provider);

            /// @dev Allow the enforcement criteria to perform any additional logic.
            require(
                !requiresSubmission || removed,
                "LaborMarket::claim: Invalid submission claim"
            );

            /// @notice Transfer the pTokens to the network participant.
            /// @dev Update health status for bulk processing offchain.
            success = request.pTokenProvider.transfer(provider, amount);
```

This prevents us from using it here in the context, as withdrawRemainder will hit ERC20 transfer amount exceeds balance if a provider has claimed their reward. (requestIdToProviders.length() == 0 if all providers claimed already and the claimAmount is equal to the initial request amount)

Reply:
Good catch, will need to make `claim` capability derived from `requestIdToAddressToPerformance` and reset them back to `0` upon completion as they have completed the full cycle and need to exit gracefully.

Reply:
Should be resolved in: d6e4c018884083a3e0c90d0c29c9f185a49e3fc0